tesseract_motion_planners/ompl/src/ompl_motion_planner.cpp

```diff
@@ -318,27 +212,25 @@ PlannerResponse OMPLMotionPlanner::solve(const PlannerRequest& request) const
 
     // Get end state kinematics data
     tesseract_common::ManipulatorInfo end_mi = composite_mi.getCombined(end_move_instruction.getManipulatorInfo());
-    tesseract_kinematics::JointGroup::UPtr manip = request.env->getJointGroup(end_mi.manipulator);
 
     // Create problem data
     const auto& start_move_instruction = start_instruction.get().as<MoveInstructionPoly>();
     std::unique_ptr<OMPLSolverConfig> solver_config = cur_plan_profile->createSolverConfig();
-    OMPLStateExtractor extractor = cur_plan_profile->createStateExtractor(*manip);
     std::shared_ptr<ompl::geometric::SimpleSetup> simple_setup;
 
-    if (cached_simple_setups.empty() || segment > cached_simple_setups.size())
+    if (cached_simple_setups.empty() || i > cached_simple_setups.size())
     {
       simple_setup =
           cur_plan_profile->createSimpleSetup(start_move_instruction, end_move_instruction, composite_mi, request.env);
       cached_simple_setups.push_back(simple_setup);
     }
     else
     {
-      simple_setup = cached_simple_setups.at(segment - 1);
+      simple_setup = cached_simple_setups.at(i - 1);
     }
 
     // Parallel Plan problem
-    auto status = parallelPlan(*simple_setup, *solver_config, num_output_states);
+    auto status = parallelPlan(*simple_setup, *solver_config, 2);
```

Comment:
Same comment about `num_output_states`. It is initialized to 1, incremented by 1 and reset to 1 in every loop, making its value 2 every time it gets to this point.